lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -521,9 +519,17 @@ def test_user_groups(self):
         mock_user.is_authenticated.return_value = False
         self.assertEqual(views.user_groups(mock_user), [])
 
+    # TODO: TNL-6546: Remove decorator for unified_course_view
+    @override_flag('unified_course_view', active=True)
     def test_get_redirect_url(self):
-        self.assertIn(
-            'activate_block_id',
+        # test the course location
+        self.assertEqual(
+            '/courses/org.0/course_0/Testing_course/courseware?activate_block_id=i4x%3A%2F%2Forg.0%2Fcourse_0%2Fcourse%2FTesting_course',
+            get_redirect_url(self.course_key, self.course.location),
+        )
+        # test a section location
+        self.assertEqual(
+            '/courses/org.0/course_0/Testing_course/courseware/Chapter_1/Sequential_1/?activate_block_id=i4x%3A%2F%2Forg.0%2Fcourse_0%2Fsequential%2FSequential_1',
```

Comment:
Also, @andy-armstrong, this passed locally before, but failed here.  I don't know why.  I tried changing to a unicode string to see if that solves the issue.